src/Stratis.Features.FederatedPeg/TargetChain/WithdrawalTransactionBuilder.cs

```diff
@@ -90,10 +92,10 @@ public Transaction BuildWithdrawalTransaction(uint256 depositId, uint blockTime,
                 {
                     this.logger.LogError("Could not create transaction for deposit {0}: {1}", depositId, error.Message);
                 }
-            }
 
-            this.logger.LogTrace("(-)[FAIL]");
-            return null;
+                this.logger.LogTrace("(-)[FAIL]");
+                return null;
+            }
```

Comment:
This return seems to have moved inside the catch block.

Reply:
Reverted.